src/mozillavpn.cpp

```diff
@@ -394,13 +394,23 @@ void MozillaVPN::maybeStateMain() {
   SettingsHolder* settingsHolder = SettingsHolder::instance();
 
 #if !defined(MZ_ANDROID) && !defined(MZ_IOS)
-  if (!settingsHolder->postAuthenticationShown()) {
+  if (Feature::get(Feature::Feature_newOnboarding)->isSupported()) {
+    if (!settingsHolder->onboardingCompleted()) {
+      if (!settingsHolder->onboardingStarted()) {
+        settingsHolder->setGleanEnabled(false);
```

Comment:
This line could use a comment that this is because the user hasn't opted into telemetry yet.